gateway/server.go

```diff
@@ -694,11 +720,14 @@ func generateOAuthPrefix(apiID string) string {
 // Create API-specific OAuth handlers and respective auth servers
 func (gw *Gateway) addOAuthHandlers(spec *APISpec, muxer *mux.Router) *OAuthManager {
 
-	apiAuthorizePath := "/tyk/oauth/authorize-client{_:/?}"
-	clientAuthPath := "/oauth/authorize{_:/?}"
-	clientAccessPath := "/oauth/token{_:/?}"
-	revokeToken := "/oauth/revoke"
-	revokeAllTokens := "/oauth/revoke_all"
+	prefixedWithListenPath := func(endpoint string) string {
```

Comment:
may as well just name it `p` or some short var - does the long function name add to readability here?